lms/djangoapps/grades/new/subsection_grade.py

```diff
@@ -228,42 +228,40 @@ def _get_saved_grade(self, subsection, course_structure, course):  # pylint: dis
                 log.warning(
-                    "Loaded saved grade for course with id {0}, version {1}, subsection {2}. grade: {3}".format(
+                    "Persistent Grades: Loaded saved grade for course with id {0}, version {1}, subtree edited on {2}, "
+                    "grade: {3}".format(
                         unicode(course.id),
-                        unicode(course.course_version),
-                        unicode(subsection.location),
-                        subsection_grade.grade_data_as_unicode()
+                        unicode(getattr(course, 'course_version', None)),
+                        unicode(course.subtree_edited_on),
+                        unicode(subsection_grade)
                     )
                 )
                 return subsection_grade
             except PersistentSubsectionGrade.DoesNotExist:
-                log.warning("Could not find saved grade for course with id {0}, version {1}, subsection {2}.".format(
-                    unicode(course.id),
-                    unicode(course.course_version),
-                    unicode(subsection.location),
-                ))
+                log.warning(
+                    "Persistent Grades: Could not find saved grade for course with id {0}, version {1}, subtree edited"
+                    " on {2}, subsection {3}.".format(
+                        unicode(course.id),
+                        unicode(getattr(course, 'course_version', None)),
+                        unicode(course.subtree_edited_on),
+                        unicode(subsection.location),
+                    )
+                )
                 return None
 
     def _save_grade(self, subsection_grade, subsection, course):  # pylint: disable=unused-argument
         """
         Updates the saved grade for the student and subsection.
         """
-        log.info("attempting to save grade for course {0}, version {1}, subsection {2}. computed grade: {3}".format(
-            unicode(course.id),
-            unicode(course.course_version),
-            unicode(subsection.location),
-            subsection_grade.grade_data_as_unicode()
-        ))
-
         if PersistentGradesEnabledFlag.feature_enabled(course.id):
             subsection_grade.save(self.student, subsection, course)
-
-        log.warning("saved grade for course {0}, version {1}, subsection {2}. computed grade: {3}".format(
-            unicode(course.id),
-            unicode(course.course_version),
-            unicode(subsection.location),
-            subsection_grade.grade_data_as_unicode()
-        ))
+        log.warning("Persistent Grades: Saved grade for course {0}, version {1}, subtree_edited_on {2}, grade: {3}"
+                    .format(
+                        unicode(course.id),
+                        unicode(getattr(course, 'course_version', None)),
+                        unicode(course.subtree_edited_on),
+                        unicode(subsection_grade)
+                    ))
```

Comment:
That said, It looks like the logging module also coerces everything to strings, so we'd lose our good unicode handling.  I don't think it's worth it.